components/dashboard/Switcher.tsx

```diff
@@ -0,0 +1,258 @@
+              setExpanded(!expanded);
+            }}
+            flexShrink={0}
+          >
+            {expanded ? <ChevronUp size="16px" /> : <ChevronDown size="16px" />}
+          </StyledRoundButton>
+        )}
+      </StyledMenuEntry>
+      {expanded &&
+        account?.children?.map(child => (
+          <StyledMenuEntry
+            key={child?.id}
+            href={`/dashboard/${child.slug}`}
+            title={child.name}
+            $isActive={activeSlug === child.slug}
+          >
+            <Flex overflow="hidden" alignItems="center" gridGap="12px">
+              <Avatar ml={4} collective={child} size={32} />
+              <Span truncateOverflow>{child.name}</Span>
+            </Flex>
+          </StyledMenuEntry>
+        ))}
+    </React.Fragment>
+  );
+};
+
+MenuEntry.propTypes = {
+  account: PropTypes.object,
+  activeSlug: PropTypes.string,
+};
+
+const Switcher = () => {
+  const { LoggedInUser } = useLoggedInUser();
+  const intl = useIntl();
+  const router = useRouter();
+  const { slug } = router.query;
+  const activeSlug = Array.isArray(slug) ? slug[0] : slug;
+
+  const loggedInUserCollective = LoggedInUser?.collective;
+
+  const allMemberships = getAllAdminMemberships(LoggedInUser);
+  const rootActiveMemberships = getAdminMemberships(LoggedInUser);
+  const rootArchivedMemberships = getAdminMembershipsForArchived(LoggedInUser);
+  const groupedMemberships = groupBy(rootActiveMemberships, m => m.collective.type);
+  groupedMemberships.ARCHIVED = rootArchivedMemberships;
+
+  const activeAccount = allMemberships.find(m => m.collective.slug === slug)?.collective || loggedInUserCollective;
+
+  return (
+    <Dropdown trigger="click">
+      {({ triggerProps, dropdownProps }) => (
+        <React.Fragment>
+          <Flex alignItems="center">
+            <DropdownButton {...triggerProps}>
+              <div>
+                <Avatar collective={activeAccount} size={32} />
+                <Span truncateOverflow>{activeAccount?.name}</Span>
+              </div>
+
+              <ChevronUpDown size={20} />
+            </DropdownButton>
+          </Flex>
+          <Container maxWidth={'100%'} {...dropdownProps}>
+            <StyledDropdownContent>
+              <Flex p={2} flexDirection="column" gridGap={3}>
+                <StyledMenuEntry
+                  key={loggedInUserCollective?.id}
+                  href={`/dashboard/${loggedInUserCollective?.slug}`}
+                  title={loggedInUserCollective?.name}
+                  $isActive={slug === loggedInUserCollective?.slug}
+                >
+                  <Flex alignItems="center" gridGap="12px">
+                    <Avatar collective={loggedInUserCollective} size={32} />
+                    <Span truncateOverflow>{loggedInUserCollective?.name}</Span>
+                  </Flex>
+                </StyledMenuEntry>
+                {Object.entries(groupedMemberships).map(([collectiveType, memberships]) => {
```

Comment:
We should filter out groups without memberships. This can result in an empty Archived group.